### PR TITLE
Add `is_transforming_instant_article()` conditional

### PR DESCRIPTION
**Pull request by goldenapples**
Adds a new function in the global namespace to check if the current view
being processed is for an instant article. Handles both the RSS feed and
content which is being processed for the Instant_Articles_Publisher
class. This conditional can be checked in any filters on 'the_content',
'the_title' or elsewhere that need to treat Instant Articles content
separately from web content.

See #426

### CODE REVIEW

**Comment by goldenapples** [comment]
I know that this logic could easily be done in a theme or another plugin that interfaces with this, now that the two actions are available, but since this has been a common need of mine, it would be nice to have something like this available directly in this plugin itself.

File: facebook-instant-articles.php [hunk 1]
```diff
@@ -157,6 +157,23 @@ function instant_articles_feed() {
 
 	}
 
+	/**
+	 * Whether currently processing an instant article.
+	 *
+	 * @param bool Set the status
+	 * @return bool
+	 */
+	function is_instant_article( $set_status = null ) {
```
Comment (diegoquinteiro) [inline facebook-instant-articles.php]: what about `is_transforming_instant_article`?

**Comment by rinatkhaziev** [comment]
@goldenapples in f81998e you attach `is_instant_article` to `before..` and `after...`, this is potentially a point of breakage since hooks can be manipulated outside the context of the plugin. I feel like it's safer to put calls inside `Instant_Articles_Post::to_instant_article`  directly. 

cc @diegoquinteiro

File: facebook-instant-articles.php (hunk 1 above)
Comment (goldenapples) [inline facebook-instant-articles.php]: Good sugeestion. I've updated the method name in aa33ecf.

**Comment by goldenapples** [comment]
@rinatkhaziev Good call. I was trying to port over the ad hoc solution I was using in a theme to the plugin. But if this method is going into the plugin directly, it does make much more sense to set the status directly rather than plugging in to those hooks. Does ca03891 look like what you had in mind?

**Comment by rinatkhaziev** [comment]
@diegoquinteiro This looks good to me, should we merge it to master and then I'm going to fix my PR?

**Comment by diegoquinteiro** [comment]
@rinatkhaziev done! =)

**Pull request merged**